### PR TITLE
feat: add support for code_verifier in getToken

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Addresses #148.  We should land #215 first.

### CODE REVIEW

File: examples/oauth2-codeVerifier.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    const oAuth2Client = await getAuthenticatedClient();
+    // Make a simple request to the Google Plus API using our pre-authenticated client. The `request()` method
+    // takes an AxiosRequestConfig object.  Visit https://github.com/axios/axios#request-config.
+    const url = 'https://www.googleapis.com/plus/v1/people?query=pizza';
+    const res = await oAuth2Client.request({url})
+    console.log(res.data);
+  } catch (e) {
+    console.error(e);
+  }
+  process.exit();
```
Comment (ofrobots) [inline examples/oauth2-codeVerifier.js]: Why is this needed?

File: examples/oauth2-codeVerifier.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+async function main() {
+  try {
+    const oAuth2Client = await getAuthenticatedClient();
+    // Make a simple request to the Google Plus API using our pre-authenticated client. The `request()` method
+    // takes an AxiosRequestConfig object.  Visit https://github.com/axios/axios#request-config.
+    const url = 'https://www.googleapis.com/plus/v1/people?query=pizza';
+    const res = await oAuth2Client.request({url})
+    console.log(res.data);
+  } catch (e) {
+    console.error(e);
```
Comment (ofrobots) [inline examples/oauth2-codeVerifier.js]: Use `main().catch(console.error)` instead of a try block here.

Comment (JustinBeckwith) [reply examples/oauth2-codeVerifier.js]: fixed!

File: examples/oauth2-codeVerifier.js (hunk 1 above)
Comment (JustinBeckwith) [reply examples/oauth2-codeVerifier.js]: fixed!

**Comment by JustinBeckwith** [comment]
*ping* :)

File: examples/oauth2-codeVerifier.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+'use strict';
+
+const {OAuth2Client} = require('../build/src/index');
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const opn = require('opn');
+
+// Download your OAuth2 configuration from the Google
```
Comment (ofrobots) [inline examples/oauth2-codeVerifier.js]: 'the Google'?

File: examples/oauth2-codeVerifier.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+const keys = require('./oauth2.keys.json');
+
+/**
+ * Start by acquiring a pre-authenticated oAuth2 client.
+ */
+async function main() {
+  const oAuth2Client = await getAuthenticatedClient();
+  // Make a simple request to the Google Plus API using our pre-authenticated client. The `request()` method
```
Comment (ofrobots) [inline examples/oauth2-codeVerifier.js]: Is clang-format being applied to this directory? This line seems overlong upon eyeballing.

File: examples/oauth2-codeVerifier.js (hunk 3 above)
Comment (JustinBeckwith) [reply examples/oauth2-codeVerifier.js]: The Goog yo :)  Half finished sentence, fixed!

File: examples/oauth2-codeVerifier.js (hunk 4 above)
Comment (JustinBeckwith) [reply examples/oauth2-codeVerifier.js]: Nope!  I opened #228 to track getting lint and formatting rules applied to the JavaScripts.
